nexus/src/db/datastore.rs

```diff
@@ -1043,6 +1046,76 @@ impl DataStore {
         }
     }
 
+    /*
+     * Network interfaces
+     */
+
+    /**
+     * Generate a unique MAC address for an interface
+     */
+    pub fn generate_mac_address(&self) -> Result<db::model::MacAddr, Error> {
+        // Bitmasks defined in RFC7042 section 2.1
+        const LOCALLY_ADMINISTERED: u8 = 0b1;
+        const MULTICAST: u8 = 0b10;
+
+        // Generate a unique address from the locally administered space.
+        use rand::Fill;
+        let mut addr = [0u8; 6];
+        addr.try_fill(&mut StdRng::from_entropy())
+            .map_err(|_| Error::internal_error("failed to generate MAC"))?;
+        addr[0] = (addr[0] & !MULTICAST) | LOCALLY_ADMINISTERED;
+        Ok(MacAddr(macaddr::MacAddr6::from(addr)).into())
+    }
```

Comment:
We can use the Oxide OUI for this purpose, as described in [RFD 174 §3.2](https://rfd.shared.oxide.computer/rfd/0174#_mac_addresses_2). That is, the space available between `A8:40:25:F0:00:00` and `A8:40:25:FF:FF:FF`. The RFD goes into some discussion about the scope of mac address reuse and such. For now we might just randomly generate the last 3 bytes in that acceptable range and then come back around to properly assigning the MAC address space per *insert scope here*.

Reply:
ah! I knew about the OUI but didn't know we had written things about the allocation of the space. Thank you

Reply:
No worries, easy to miss. And I think your question on the RFD is a fair one. One nice thing about using the VM subspace of the global address is that it's a bit more obvious during debugging that a given mac does indeed relate to an Oxide instance, and not something else.